### PR TITLE
Split fake RP into its own process

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/openshift-azure/issues/912

### CODE REVIEW

File: cmd/fakerp/main.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+func main() {
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	log := logrus.NewEntry(logrus.StandardLogger())
+
+	// TODO: Validate all required env variables are in place
+
+	log.Info("starting the fake resource provider")
+	s := fakerp.NewServer(log, os.Getenv("RESOURCEGROUP"), "localhost:8080")
```
Comment (thekad) [inline cmd/fakerp/main.go]: Does this have to be hardcoded on port 8080? Can't the port be randomized in one of the upper ephemeral ports for the duration of the run?

File: cmd/recoveretcdcluster/recoveretcdcluster.go [hunk 2]
```diff
@@ -22,15 +23,11 @@ func main() {
...
-	conf, err := fakerp.NewConfig(log, true)
-	if err != nil {
-		log.Fatal(err)
-	}
 
 	ctx := context.Background()
-	ctx = context.WithValue(ctx, api.ContextKeyClientID, conf.ClientID)
-	ctx = context.WithValue(ctx, api.ContextKeyClientSecret, conf.ClientSecret)
-	ctx = context.WithValue(ctx, api.ContextKeyTenantID, conf.TenantID)
+	ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
```
Comment (thekad) [inline cmd/recoveretcdcluster/recoveretcdcluster.go]: I sort of feel this is going in the wrong direction, maybe it should mirror how client.go loads its configuration via envconfig? Is there any hard reason why these are environment variables now?

File: hack/delete.sh [hunk 3]
```diff
@@ -14,6 +14,8 @@ fi
 USE_PROD_FLAG="-use-prod=false"
 if [[ -n "$TEST_IN_PRODUCTION" ]]; then
     USE_PROD_FLAG="-use-prod=true"
+else
```
Comment (thekad) [inline hack/delete.sh]: if you're adding `go generate` in create.sh, you should add it here as well

Comment (thekad) [reply hack/delete.sh]: Actually, this should point to `hack/fakerp.sh` I think

File: hack/create.sh [hunk 4]
```diff
@@ -12,10 +12,13 @@ mkdir -p _data/_out
 
 set -x
 
+go generate ./...
+
 USE_PROD_FLAG="-use-prod=false"
 if [[ -n "$TEST_IN_PRODUCTION" ]]; then
     USE_PROD_FLAG="-use-prod=true"
+else
+	go run cmd/fakerp/main.go &
```
Comment (thekad) [inline hack/create.sh]: should probably point to `hack/fakerp.sh`

File: cmd/fakerp/main.go (hunk 1 above)
Comment (kargakis) [reply cmd/fakerp/main.go]: Why do we care about randomizing it?

File: cmd/recoveretcdcluster/recoveretcdcluster.go (hunk 2 above)
Comment (kargakis) [reply cmd/recoveretcdcluster/recoveretcdcluster.go]: Using something like envconfig can be a follow-up, for start I wanted to remove the unnecessary dependency.

File: hack/delete.sh (hunk 3 above)
Comment (kargakis) [reply hack/delete.sh]: done

File: hack/create.sh (hunk 4 above)
Comment (kargakis) [reply hack/create.sh]: done

File: hack/create.sh [hunk 5]
```diff
@@ -15,7 +15,8 @@ set -x
 USE_PROD_FLAG="-use-prod=false"
 if [[ -n "$TEST_IN_PRODUCTION" ]]; then
     USE_PROD_FLAG="-use-prod=true"
+else
+	hack/fakerp.sh $RESOURCEGROUP &
 fi
 
-go generate ./...
```
Comment (mjudeikis) [inline hack/create.sh]: Any particular reason this got removed?

Comment (kargakis) [reply hack/create.sh]: Runs as part of `fakerp.sh` now.

Comment (mjudeikis) [reply hack/create.sh]: I would like this better in main 'root' script at the top somewhere?

Comment (kargakis) [reply hack/create.sh]: What do you mean by root script? hack/create.sh? Generation is related to the backend code run by the fake RP so it's natural place is fakerp.sh.

File: hack/create.sh [hunk 6]
```diff
@@ -15,7 +15,8 @@ set -x
 USE_PROD_FLAG="-use-prod=false"
 if [[ -n "$TEST_IN_PRODUCTION" ]]; then
     USE_PROD_FLAG="-use-prod=true"
+else
+	hack/fakerp.sh $RESOURCEGROUP &
```
Comment (kargakis) [inline hack/create.sh]: Unfortunately this is leaking and needs to be cleaned up.

Comment (kargakis) [reply hack/create.sh]: https://github.com/openshift/openshift-azure/issues/924

**Pull request merged**